calico_mesos/calico_mesos.py

```diff
@@ -493,8 +509,41 @@ def allocate(args):
         except TypeError:
             raise IsolatorException("num_ipv6 must be an integer")
 
-    return _allocate(num_ipv4, num_ipv6, hostname, uid)
-
+    # Check if the user has requested a specific IP via label
+    # Note: this will be deprecated once marathon provides the ipv4_addrs   
+    # field which will trigger netmodules' 'reserve'.
+    # This implementation replaces the requested IPAM IP with the static IP.
+    # i.e. If a user requests 1 IP, and specifies one specific IP in the 
+    # ipv4_addr label, they will receive 1 IP back - the one they specified.
```

Comment:
Haven't we already told netmodules which IP we are using by this point?  Doesn't that mean it thinks we're using a different IP than we actually are?

Also, what happens to the IP address that was assigned?  Doesn't that need to be released back to the pool?